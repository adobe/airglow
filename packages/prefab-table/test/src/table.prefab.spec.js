/*
 * ************************************************************************
 * ADOBE CONFIDENTIAL
 * ___________________
 *
 *   Copyright 2017 Adobe Systems Incorporated
 *   All Rights Reserved.
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of Adobe Systems Incorporated and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to Adobe Systems Incorporated and its
 * suppliers and are protected by all applicable intellectual property
 * laws, including trade secret and copyright laws.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from Adobe Systems Incorporated.
 * ************************************************************************
 */

import tablePrefab from '../../src/table.prefab';

const dataSelector = state => state;

const columns = {
  name: {
    sorter: 'defalut'
  },
  color: {
    sorter: () => () => -1
  }
};

describe('tablePrefab', () => {
  it('should return the table', function () {
    expect(tablePrefab({ name: 'fruits', dataSelector, columns })).toMatchSnapshot();
  });
});
