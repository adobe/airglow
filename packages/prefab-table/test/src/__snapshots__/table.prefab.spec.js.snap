// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tablePrefab should return the table 1`] = `
Object {
  "fruits": Object {
    "addColumnAction": [Function],
    "allowsMultipleSelection": true,
    "allowsSelection": true,
    "columns": Object {
      "color": Object {
        "name": "color",
        "sorter": [Function],
      },
      "name": Object {
        "name": "name",
        "sorter": [Function],
      },
    },
    "dataSelector": [Function],
    "defaultSort": undefined,
    "deselectRowsAction": [Function],
    "handlers": [Function],
    "reduce": [Function],
    "removeColumnAction": [Function],
    "selectedData": [Function],
    "state": [Function],
    "tableName": "fruits",
    "toggleRowAction": [Function],
  },
}
`;
